### PR TITLE
[backport] fix sanitation

### PR DESCRIPTION
**Pull request by bkcsoft**
Backport of #1461

### CODE REVIEW

File: modules/markdown/sanitizer.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	init   sync.Once
+}
+
+var sanitizer = &Sanitizer{}
+
+// NewSanitizer initializes sanitizer with allowed attributes based on settings.
+// Multiple calls to this function will only create one instance of Sanitizer during
+// entire application lifecycle.
+func NewSanitizer() {
+	log.Trace("Markdown: sanitizer initialization requested")
```
Comment (lunny) [inline modules/markdown/sanitizer.go]: I remember this will break SSH

File: modules/markdown/sanitizer.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		sanitizer.policy.AllowAttrs("class").Matching(regexp.MustCompile(`^language-\w+$`)).OnElements("code")
+
+		// Checkboxes
+		sanitizer.policy.AllowAttrs("type").Matching(regexp.MustCompile(`^checkbox$`)).OnElements("input")
+		sanitizer.policy.AllowAttrs("checked", "disabled").OnElements("input")
+
+		// Custom URL-Schemes
+		sanitizer.policy.AllowURLSchemes(setting.Markdown.CustomURLSchemes...)
+
+		log.Trace("Markdown: sanitizer initialized")
```
Comment (lunny) [inline modules/markdown/sanitizer.go]: Also this line.

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM

**Review by lunny — CHANGES_REQUESTED** [review]

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
LGTM

**Pull request merged**